### PR TITLE
support for commitByDefault

### PR DESCRIPTION
**Pull request by Ashok-Varma**
Thankyou for such a wonderful library. Using it in many of my projects.

in many cases app might need **commit** for some properties. 
> commit/apply doesn't much depend on scenario, those rather depend on property.
> we can use blockingCommit but adding blockingCommit every time we access that property is difficult.

**Changes**
1. added support for commitByDefault
2. added Parameterized unit tests for commitByDefault feature
3. no breaking changes

### CODE REVIEW

**Comment by Ashok-Varma** [comment]
@chibatching failed due to kotlin incremental builds.

> extension function added in kotPref module. :enum-support:compileDebugKotlin is executed first and i think this is using cached version of kotPref so it can't find the extension function.

may be building and testing kotPref module before the enum-support module might help

File: kotpref/src/main/kotlin/com/chibatching/kotpref/SharedPrefExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Extension to choose between {@link SharedPreferences.Editor.commit} and {@link SharedPreferences.Editor.apply}
+ * @param synchronous save to sharedPref file instantly
+ */
+fun SharedPreferences.Editor.execute(synchronous: Boolean) {
+    if (synchronous)
+        commit()
+    else
+        apply()
```
Comment (chibatching) [inline kotpref/src/main/kotlin/com/chibatching/kotpref/SharedPrefExtensions.kt]: I prefer to if-else with one line or braces block, according to kotlin android style guide.
Could you fix it?

ref: https://android.github.io/kotlin-guides/style.html#braces

**Comment by chibatching** [comment]
@chibatching TODO investigate CI failed issue

File: kotpref/src/main/kotlin/com/chibatching/kotpref/SharedPrefExtensions.kt (hunk 1 above)
Comment (Ashok-Varma) [reply kotpref/src/main/kotlin/com/chibatching/kotpref/SharedPrefExtensions.kt]: @chibatching thank you, made the required changes.

**Comment by chibatching** [comment]
@Ashok-Varma I updated android gradle plugin on master. It seems to fix ci issue.
Please rebase you PR 😃

**Comment by Ashok-Varma** [comment]
@chibatching all green 👍

**Comment by chibatching** [comment]
Thanks!! 😄

**Pull request merged**